src/dxvk/dxvk_image.cpp

```diff
@@ -83,10 +94,35 @@ namespace dxvk {
     dedMemoryAllocInfo.pNext  = VK_NULL_HANDLE;
     dedMemoryAllocInfo.buffer = VK_NULL_HANDLE;
     dedMemoryAllocInfo.image  = m_image.image;
-    
+
+    VkExportMemoryAllocateInfo exportInfo;
+    if (m_shared && createInfo.sharing.isExport()) {
+      exportInfo.sType = VK_STRUCTURE_TYPE_EXPORT_MEMORY_ALLOCATE_INFO;
+      exportInfo.pNext = nullptr;
+      exportInfo.handleTypes = createInfo.sharing.type;
+
+      dedMemoryAllocInfo.pNext = &exportInfo;
+    }
+
+    VkImportMemoryWin32HandleInfoKHR importInfo;
+    if (m_shared && createInfo.sharing.isImport()) {
+      importInfo.sType = VK_STRUCTURE_TYPE_IMPORT_MEMORY_WIN32_HANDLE_INFO_KHR;
+      importInfo.pNext = nullptr;
+      importInfo.handleType = createInfo.sharing.type;
+      importInfo.handle = createInfo.sharing.handle;
+      importInfo.name = nullptr;
+
+      dedMemoryAllocInfo.pNext = &importInfo;
+    }
+
     m_vkd->vkGetImageMemoryRequirements2(
       m_vkd->device(), &memReqInfo, &memReq);
 
+    if (m_shared) {
+      dedicatedRequirements.prefersDedicatedAllocation  = VK_TRUE;
+      dedicatedRequirements.requiresDedicatedAllocation = VK_TRUE;
+    }
```

Comment:
I think the spec specifies it as a "may"

https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkMemoryDedicatedRequirementsKHR.html

> requiresDedicatedAllocation may be VK_TRUE if the pNext chain of [VkImageCreateInfo](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkImageCreateInfo.html) for the call to [vkCreateImage](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkCreateImage.html) used to create the image being queried included a [VkExternalMemoryImageCreateInfo](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkExternalMemoryImageCreateInfo.html) structure, and any of the handle types specified in [VkExternalMemoryImageCreateInfo](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkExternalMemoryImageCreateInfo.html)::handleTypes requires dedicated allocation, as reported by [vkGetPhysicalDeviceImageFormatProperties2](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkGetPhysicalDeviceImageFormatProperties2.html) in VkExternalImageFormatProperties::externalMemoryProperties.externalMemoryFeatures. Otherwise, requiresDedicatedAllocation will be VK_FALSE.

Reply:
Yes, so if the driver needs it (which they all do), it will set it.